### DISCUSSION TITLE
Prompt Module Presedence

### DISCUSSION THREAD

**Discussion by bretonics** [starship/starship#2641] [Q&A] [answered]
Hi,

I've been playing around with Starship and other prompt frameworks, specifically [Spaceship](https://denysdovhan.com/spaceship-prompt/), and evaluating the features and customization. I am leaning more towards Starship thanks to it's cross-platform capabilities and extensible customization. Thank you!

Now, this is part question, maybe part feature request....

I have noticed that Module contextual information seems to have precedence and may override other modules?? But I don't see any information in the docs about this. For instance, I had just updated Docker and so decided to go through their [getting started tutorial](https://github.com/docker/getting-started), which downloads a repo with a Dockerfile and a requirements.txt (Python) file. I noticed that only the Python contextual information showed in the prompt; I was expecting Docker. Interestingly though, Spaceship detected and displayed the Docker information rather than Python (what I initially expected). So, why is this the case?

https://user-images.githubusercontent.com/3980538/115960007-17441e80-a4dd-11eb-8961-5306e396a6d9.mov

Does one module have precedence over others?
Which one shows up when 2+ modules have contextual information in the same directory like this?

I thought it might have to do with the [format](https://starship.rs/config/#prompt) order, but I tried switching the order and that didn't work. Looks like others don't clash, such as git and Python, but why is this being observed with Docker?

Thanks for the additional information.

**Comment by andytom** [comment, marked as answer]
Starship does not have a precedent, it will show all modules that are active. The difference is that the starship [docker_context module](https://starship.rs/config/#docker-context) only shows the [docker context](https://docs.docker.com/engine/context/working-with-contexts/) if it is not default.

**Reply by bretonics** [reply]
Thank you. That makes more sense. Is there a way to always enable it, i.e. if the docker context is not the default?

**Reply by andytom** [reply]
I think if you set 
```toml
[docker_context] 
only_with_files = false
```

It should show the docker context everywhere (so long as it is not default).
